README.md

```diff
@@ -1,2 +1,311 @@
+
+> A test suite and interface you can use to implement the IPFS Core API.
+
+The primary goal of this module is to define and ensure that both IPFS core implementations and their respective HTTP client libraries offer the same interface, so that developers can quickly change between a local and a remote node without having to change their applications. In addition to the definition of the expected interface, this module offers a suite of tests that can be run in order to check if the interface is used as described.
+
+The API is presented with both Node.js and Go primitives, however, there is not actual limitations for it to be extended for any other language, pushing forward the cross compatibility and interop through diferent stacks.
+
+# Modules that implement the interface
+
+- [JavaScript IPFS implementation](https://github.com/ipfs/js-ipfs)
+- [JavaScript ipfs-api](https://github.com/ipfs/js-ipfs-api)
+- Soon, go-ipfs, go-ipfs-api, java-ipfs-api, python-ipfs-api and others will implement it as well.
+
+Send a PR to add a new one if you happen to find or write one.
+
+# Badge
+
+Include this badge in your readme if you make a new module that uses interface-stream-muxer API.
+
+![](/img/badge.png)
+
+# How to use the battery tests
+
+## Node.js
+
+Install interface-ipfs-core as one of the dependencies of your project and as a test file, using `mocha` (for Node.js) or a test runner with compatible API, do:
+
+```
+var test = require('interface-ipfs-core')
+
+var common = {
+  setup: function (cb) {
+    cb(null, yourIPFSInstance)
+  },
+  teardown: function (cb) {
+    cb()
+  }
+}
+
+// use all of the test suits
+test.all(common)
+```
+
+## Go
+
+> WIP
+
+# API
+
+A valid (read: that follows this interface) IPFS core implementation, must expose the following API.
+
+## Object
+
+### `object.new`
+
+> Create a new MerkleDAG node, using a specific layout. Caveat: So far, only UnixFS object layouts are supported.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.object.new(layout, [callback])
+
+`layout` is the MerkleDAG node type, it can be: `null`, `'unixfs-dir'`, `'unixfs-raw'`, `'unixfs-file'`, `'unixfs-metadata'`, `'unixfs-symlink'`.
+
+`callback` must follow `function (err, node) {}` signature, where `err` is an error if the operation was not successful and `node` is a MerkleDAG node of the type [DAGNode](https://github.com/vijayee/js-ipfs-merkle-dag/blob/master/src/dag-node.js)
+
+If no `callback` is passed, a promise is returned.
+
+
+
+
+
+### `object.put`
+
+> Store an MerkleDAG node.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.object.put(obj, [options, callback])
+
+`obj` is the MerkleDAG Node to be stored. Can of type:
+
+- Object, with format `{ Data: <data>, Links: [] }`
+- Buffer, requiring that the encoding is specified on the encoding
+- [DAGNode](https://github.com/vijayee/js-ipfs-merkle-dag/blob/master/src/dag-node.js)
+
```

Comment:
Yes, Yes and no, following the same case as before, `new DAGNode(new Buffer('hello'))`